web/modules/custom/custom_weather/src/Form/CustomWeatherUserForm.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Drupal\custom_weather\Form;
+
+use Drupal\Core\Database\Database;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\custom_weather\Service\UserCityHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a configuration form for managing settings related to the module.
+ */
+class CustomWeatherUserForm extends FormBase {
+
+  public function __construct(protected UserCityHandler $city) {
+    $this->databaseService = $city;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): CustomWeatherUserForm|FormBase|static {
+    return new static(
+      $container->get('Drupal\custom_weather\Service\UserCityHandler')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'custom_weather_user';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $city = $this->databaseService->getCityFromDatabase();
+    // Array of cities for the dropdown list.
+    $cities = [
+      'Kyiv' => $this->t('Kyiv'),
+      'Lviv' => $this->t('Lviv'),
+      'Rivne' => $this->t('Rivne'),
+      'Lutsk' => $this->t('Lutsk'),
+      'Zhytomyr' => $this->t('Zhytomyr'),
+      'Chernivtsi' => $this->t('Chernivtsi'),
+      'Ternopil' => $this->t('Ternopil'),
+      'Khmelnytskyi' => $this->t('Khmelnytskyi'),
+      'Uzhhorod' => $this->t('Uzhhorod'),
+      'Vinnytsia' => $this->t('Vinnytsia'),
+      'Cherkasy' => $this->t('Cherkasy'),
+      'Poltava' => $this->t('Poltava'),
+      'Chernihiv' => $this->t('Chernihiv'),
+    ];
+
+    // Add the dropdown list with cities to the form.
+    $form['selected_city'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Select a city'),
+      '#options' => $cities,
+      '#default_value' => $city,
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Form submission handler for your custom form.
+   *
+   * @throws \Exception
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $user_id = \Drupal::currentUser()->id();
+    $selected_user_city = $form_state->getValue('selected_city');
+    $connection = Database::getConnection();
+    $query = $connection->select('custom_weather_data', 'cwd');
+    $query->fields('cwd', ['user_id']);
+    $query->condition('user_id', $user_id);
+    $result = $query->execute()->fetchField();
+
+    if ($result) {
+      $connection->update('custom_weather_data')
+        ->fields([
+          'city' => $selected_user_city,
+        ])
+        ->condition('user_id', $user_id)
+        ->execute();
+    }
+    else {
+      $connection->insert('custom_weather_data')
+        ->fields([
+          'city' => $selected_user_city,
+          'user_id' => $user_id,
+        ])
+        ->execute();
```

Comment:
Why all of that not handled by UserCityHandler?

Reply:
In my module, there is the ability to change the city for the user only, so this submitForm is unique and works only for this form.